src/main/java/org/eclipse/uprotocol/uuid/factory/UUIDUtils.java

```
@@ -21,163 +21,191 @@
+        return uuid == null ? Optional.empty() : Optional.of(uuid.toString());
     }
 
     /**
      * Convert the UUID to byte array
      * @return The byte array
      */
-    public static byte[] toBytes(UUID uuid) {
+    public static Optional<byte[]> toBytes(UUID uuid) {
+        if (uuid == null) {
+            return Optional.empty();
+        }
         byte[] b = new byte[16];
-        return ByteBuffer.wrap(b).order(ByteOrder.BIG_ENDIAN).putLong(uuid.getMostSignificantBits())
-                .putLong(uuid.getLeastSignificantBits()).array();
+        return Optional.of(ByteBuffer.wrap(b).order(ByteOrder.BIG_ENDIAN).putLong(uuid.getMostSignificantBits())
+                .putLong(uuid.getLeastSignificantBits()).array());
     }
 
     /**
      * Convert the byte array to a UUID
      * @param bytes The UUID in bytes format
      * @return  UUIDv8 object built from the byte array
      */
-    public static UUID fromBytes(byte[] bytes) {
+    public static Optional<UUID> fromBytes(byte[] bytes) {
+        if (bytes == null || bytes.length != 16) {
+            return Optional.empty();
+        }
+
         final ByteBuffer byteBuffer = ByteBuffer.wrap(bytes);
-        return new UUID(byteBuffer.getLong(), byteBuffer.getLong());
+        return Optional.of(new UUID(byteBuffer.getLong(), byteBuffer.getLong()));
     }
 
+
     /**
      * Create a UUID from the passed string
      * @param string the string representation of the uuid
-     * @return The UUIDv8 representation of the string
+     * @return The UUID object representation of the string
      */
-    public static UUID fromString(String string) {
-        return UUID.fromString(string);
+    public static Optional<UUID> fromString(String string) {
+        UUID uuid = null;
+        if (string == null || string.isBlank()) {
+            return Optional.empty();
+        }
+        try {
+            uuid = UUID.fromString(string);
+        } catch (IllegalArgumentException e) {
```

you are swallowing exceptions - how will we know how to fix the problem - the problem should be at least logged